app/services/merge_patient_service.rb

```diff
@@ -1,64 +1,61 @@
 class MergePatientService
-  def initialize(payload)
+  def initialize(payload, request_metadata:)
     @payload = payload
+    @request_metadata = request_metadata
+    @existing_patient = Patient.with_discarded.find_by(id: payload["id"])
   end
 
   def merge
-    existing_patient_attributes = Patient.with_discarded.find_by(id: payload["id"])&.attributes
     merged_address = Address.merge(payload[:address]) if payload[:address].present?
 
-    patient_attributes = payload.except(:address, :phone_numbers, :business_identifiers)
+    patient_attributes =
```

Comment:
These transformations are better treated as a pipeline operations so that the order is always clear.